src/Microsoft.DotNet.ProjectModel/Compilation/LibraryExporter.cs

```diff
@@ -91,16 +93,14 @@ private IEnumerable<LibraryExport> ExportLibraries(Func<LibraryDescription, bool
                     }
                 }
 
+                // Source references are not transitive
```

Comment:
@davidfowl I considered that. One issue is that the current MSBuild NuGet tooling does transitive analyzer referencing. We could put in a depth limit but people may expect things like an analyzer "metapackage" to provide access to all the nested analyzers.